harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -437,7 +437,8 @@ def test_migrate(self, api_client, unique_vm_name, wait_timeout):
             f"{data}"
         )
 
-        new_host = next(h['id'] for h in host_data['data'] if cur_host != h['id'])
+        new_host = next(h['id'] for h in host_data['data']
+                        if cur_host != h['id'] and not h['spec'].get('taint'))
```

Comment:
It should be rare, but should we add a try-except block to log when get StopIteration when empty iterator (no available new host)?

Reply:
I think this should be fine, when `next` the iterator, **StopIteration** will raised then this line (of code) will be highlight, and the variable already explain that it is about `new_host` which is empty. (by **StopIteration**)